aten/src/ATen/native/quantized/cpu/qnnpack/src/hardsigmoid.c

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#include <assert.h>
+#include <math.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+#include <pytorch_qnnpack.h>
+#include <qnnpack/log.h>
+#include <qnnpack/operator.h>
+
+enum pytorch_qnnp_status pytorch_qnnp_create_hardsigmoid_nc_q8(
+    size_t channels,
+    uint8_t input_zero_point,
+    float input_scale,
+    uint8_t output_zero_point,
+    float output_scale,
+    uint8_t output_min,
+    uint8_t output_max,
+    uint32_t flags,
+    pytorch_qnnp_operator_t* hardsigmoid_out) {
+  pytorch_qnnp_operator_t hardsigmoid_op = NULL;
+  enum pytorch_qnnp_status status = pytorch_qnnp_status_uninitialized;
+
+  if (!pytorch_qnnp_params.initialized) {
+    pytorch_qnnp_log_error(
+        "pytorch_qnnp_create_hardsigmoid_nc_q8 failed because QNNPACK is not properly initialized");
+    goto error;
```

Comment:
`goto` is generally considered harmful: http://david.tribble.com/text/goto.html Consider replacing with 

```
pytorch_qnnp_delete_operator(hardsigmoid_op);
return pytorch_qnnp_status_uninitialized;
```